Fix build error in mac m1 silicon&Improve readme (#1031)
paritytech/parity-bridges-common: README.md

cargo build -p substrate-relay
```

### Running a Dev network

We will launch a dev network to demonstrate relay from millau to rialto here

#### Running from local scripts

To run a simple dev network you'll can use the scripts located in

> [the `deployments/local-scripts` folder](./deployments/local-scripts). Since the relayer connects to both Substrate chains it must be run last.

Reviewer: Can you make this two lines again? We wrap our files at 100 characters (see the [`.editorconfig`](https://github.com/paritytech/parity-bridges-common/blob/master/.editorconfig#L9)).

```bash

> # run nodes In **parity-bridges-common** folder

Reviewer: ```suggestion
# In `parity-bridges-common` folder
```

./deployments/local-scripts/run-rialto-node.sh
./deployments/local-scripts/run-millau-node.sh

# init bridge & relay headers from millau to rialto
./deployments/local-scripts/relay-millau-to-rialto.sh

# relay messages from millau to rialto
./deployments/local-scripts/relay-messages-millau-to-rialto.sh

> # relay messages from rialto to millau for acknowledgement to work

Reviewer: Don't think this is required, it's really  a detail that someone getting up and running doesn't need to worry about

./deployments/local-scripts/relay-messages-millau-to-rialto.sh
```

At this point you should see the relayer submitting headers from the Millau Substrate chain to the
Rialto Substrate chain as following:

```
# Header Relayer Logs

> [Millau_to_Rialto_Sync] 2021-06-27 11:08:43 +0000 DEBUG bridge Going to submit finality proof of Millau header #147 to Rialto

Reviewer: ```suggestion
[Millau_to_Rialto_Sync] <date> DEBUG bridge Going to submit finality proof of Millau header #147 to Rialto
```

The date isn't actually that important, let's snip it

[Millau_to_Rialto_Sync] 2021-06-27 11:08:48 +0000 INFO bridge Synced 147 of 147 headers
[Millau_to_Rialto_Sync] 2021-06-27 11:08:53 +0000 DEBUG bridge Going to submit finality proof of Millau header #148 to Rialto
[Millau_to_Rialto_Sync] 2021-06-27 11:08:58 +0000 INFO bridge Synced 148 of 149 headers
```

> You will also see the message lane relayers listening for new messages. To send a message see the [How to send a message section](#How to send a message).

Reviewer: ```suggestion
You will also see the message lane relayers listening for new messages. To send a message see the [How to send a message section](#how-to-send-a-message).
```

Also, can you please wrap this at 100 characters?

```
# Message Relayer Logs

> [Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:01 +0000 DEBUG bridge Asking Millau::ReceivingConfirmationsDelivery about best message nonces

Reviewer: ```suggestion
[Millau_to_Rialto_MessageLane_00000000] <date> DEBUG bridge Asking Millau::ReceivingConfirmationsDelivery about best message nonces
```

Same thing about the date here

cargo build -p substrate-relay
```

### Running dev network

> We will only launch a dev network to demonstrate the unidirectional relay from millau to rialto here

Reviewer: This isn't entirely correct. We need a bi-directional bridge in order to relay messages

#### Running from local script

To run a simple dev network you'll can use the scripts located in
[the `deployments/local-scripts` folder](./deployments/local-scripts). Since the relayer connects to
both Substrate chains it must be run last. 

```bash

# run rialto node
./deployments/local-scripts/run-rialto-node.sh

# run millau node
./deployments/local-scripts/run-millau-node.sh

> # init bridge & relay headers from millau to rialto

Reviewer: Please fix this

./deployments/local-scripts/relay-millau-to-rialto.sh

> # relay messages from millau to rialto

Reviewer: Same here

> ./deployments/local-scripts/relay-messages-millau-to-rialto.sh

Reviewer: We also need to run the other side of the message bridge (Rialto -> Millau) or else message acknowledgements won't work

```

> At this point you should see the relayer submitting headers/messages from the Millau Substrate chain to the

Reviewer: So you won't see any messages being relayed at this point. That's a later section of the `README`.

```suggestion
At this point you should see the relayer submitting headers from the Millau Substrate chain
to the
```
Maybe you can add a sentence after this saying: "You will also see the message lane relayers listening for new messages. To send a message see the `How to send a message` section.

Rialto Substrate chain as following:

> ``` from relay headers

Reviewer: ```suggestion

# Header Relayer Logs
```

GitHub is rendering this a bit strange, but leave comment instead of annotating the ticks.

[Millau_to_Rialto_Sync] 2021-06-27 11:08:43 +0000 DEBUG bridge Going to submit finality proof of Millau header #147 to Rialto
[Millau_to_Rialto_Sync] 2021-06-27 11:08:48 +0000 INFO bridge Synced 147 of 147 headers
[Millau_to_Rialto_Sync] 2021-06-27 11:08:53 +0000 DEBUG bridge Going to submit finality proof of Millau header #148 to Rialto
[Millau_to_Rialto_Sync] 2021-06-27 11:08:58 +0000 INFO bridge Synced 148 of 149 headers
```

> ``` from relay messages

Reviewer: Same thing here, turn it into a comment instead

[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:01 +0000 INFO bridge Synced Some(2) of Some(3) nonces in Millau::MessagesDelivery -> Rialto::MessagesDelivery race
[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:01 +0000 DEBUG bridge Asking Millau::MessagesDelivery about message nonces
[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:01 +0000 DEBUG bridge Received best nonces from Millau::ReceivingConfirmationsDelivery: TargetClientNonces { latest_nonce: 0, nonces_data: () }
[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:01 +0000 DEBUG bridge Asking Millau::ReceivingConfirmationsDelivery about finalized message nonces
[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:01 +0000 DEBUG bridge Received finalized nonces from Millau::ReceivingConfirmationsDelivery: TargetClientNonces { latest_nonce: 0, nonces_data: () }
[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:01 +0000 DEBUG bridge Received nonces from Millau::MessagesDelivery: SourceClientNonces { new_nonces: {}, confirmed_nonce: Some(0) }
[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:06 +0000 DEBUG bridge Asking Millau node about its state

> [Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:06 +0000 DEBUG bridge Received state from Millau node: ClientState { best_self: HeaderId(1593, 0xacacac493e81123007ea0f6d869640b0f10e62215bc07ec234e7f3cd52f8549be95ed422efa3b48fae1ea3f6f2b672dce9c6eeb3035772758e9cda4412375277), best_finalized_self: HeaderId(1590, 0x0be81d0bc399585825f22902a396f2c3efb2e33e7e3ae6ff5f2de771a81cd5bf0ff914fb0bcccf9ef218e58799c5509032b7a1be368f8edb604e0207ea9334af), best_finalized_peer_at_best_self: HeaderId(0, 0xdcdd89927d8a348e00257e1ecc8617f45edb5118efff3ea2f9961b2ad9b7690a) }

Reviewer: ```suggestion
[Millau_to_Rialto_MessageLane_00000000] 2021-06-27 14:40:06 +0000 DEBUG bridge Received state from Millau node: ClientState { best_self: HeaderId(1593, 0xac...77), best_finalized_self: HeaderId(1590, 0x0b...af), best_finalized_peer_at_best_self: HeaderId(0, 0xdc...0a) }
```

The full hashes don't add much and they make the logs hard to read

### How to send a message

> In this section we'll show you how to quickly send a bridge message, if you want to interact with and test the bridge see more details in [send message](./docs/send-message.md)

Reviewer: Same comment about 100 character lines

```bash

> 

Reviewer: ```suggestion
```

> # send remark message from millau to rialto

Reviewer: The script name already says what this does, the comment isn't useful

./scripts/send-message-from-millau-rialto.sh remark
```

> you will see logs as following showing transaction successfully sent

Reviewer: ```suggestion
After sending a message you will see the following logs showing the transaction was successfully sent:
```

docker run -p 30334:30333 -p 9934:9933 -p 9945:9944 \
           -it paritytech/millau-bridge-node --dev --tmp \
           --rpc-cors=all --unsafe-rpc-external --unsafe-ws-external

docker run --network=host -it \

>         paritytech/substrate-relay init-bridge MillauToRialto \

Reviewer: Why are you switching this?

        --target-host localhost \
        --target-port 9944 \
        --source-host localhost \
        --source-port 9945 \
        --target-signer //Alice

docker run --network=host -it \
        paritytech/substrate-relay relay-headers MillauToRialto \
        --target-host localhost \
        --target-port 9944 \
        --source-host localhost \
        --source-port 9945 \
        --target-signer //Bob

docker run --network=host -it \

>         paritytech/substrate-relay relay-messages MillauToRialto \

Reviewer: Same thing here, to relay messages we need the other side of the bridge running (Rialto -> Millau)

        --lane 00000000 \
        --source-host localhost \
        --source-port 9945 \
        --source-signer //Bob \
        --target-host localhost \
        --target-port 9944 \

>         --target-signer //Bob        

Reviewer: ```suggestion
        --target-signer //Bob
```

Author: move all these to the docker deploy section

        --source-port 9944 \
        --target-signer //Bob \
```

You should now see the relayer submitting headers from the Millau chain to the Rialto chain.
        --source-port 9945 \
        --target-signer //Bob

If you don't want to use the published Docker images you can build images yourself. You can do this
by running the following commands at the top level of the repository.

```bash
# In `parity-bridges-common` folder
docker build . -t local/rialto-bridge-node --build-arg PROJECT=rialto-bridge-node
docker build . -t local/millau-bridge-node --build-arg PROJECT=millau-bridge-node
docker build . -t local/substrate-relay --build-arg PROJECT=substrate-relay
docker run --network=host -it \
        paritytech/substrate-relay relay-messages MillauToRialto \
        --lane 00000000 \
        --source-host localhost \
        --source-port 9945 \
        --source-signer //Bob \
        --target-host localhost \
        --target-port 9944 \
        --target-signer //Bob        
```

_Note: Building the node images will take a long time, so make sure you have some coffee handy._

Once you have the images built you can use them in the previous commands by replacing

> `paritytech/<component_name>` with `local/<component_name>` everywhere.

Author: I think we should keep the tutorial simple/focus enough,  not nessessary to teach developers to build docker image and for production deployment  they can check detail in [docker compose deployment section]()

> You should now see the relayer submitting headers/messages from the Millau chain to the Rialto chain.

Reviewer: ```suggestion
You should now see the relayer submitting headers from the Millau chain to the Rialto chain.
```

### How to send a message

A straightforward way to interact with and test the bridge is sending messages. This is explained

> in the [send message](./docs/send-message.md) document.

Reviewer: This document still contains useful info, please leave a reference to it.

You can expand on this by saying something like: In this section we'll show you how to quickly send a bridge message, if you want to see more details read the "send message" docs.

> # send remark message from 

Reviewer: ```suggestion
```

The script name says this already

./scripts/send-message-from-millau-rialto.sh remark
```

> you will see logs as following showing transaction successfully sent to Millau node

Reviewer: ```suggestion
You will see logs as following showing transaction successfully sent to Millau node:
```

```
...
INFO bridge Sending message to Rialto. Size: 286. Dispatch weight: 1038000. Fee: 275,002,568

> INFO bridge Signed Millau Call: 0x7904710484d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d0148b5009cfb8d63dc0aa61308f6ef85a570b94ad313425d5d6212e2309d9dd223b397742caed45e031e512b79fd225e3f298acced9841d381162583e9a8c6418e000c0000030000000001000000b0d60f000000000001d43593c715fdd31c61141abd04a99fd6822c8558854ccde39a5684e7a56da27d018eaf04151687736326c9fea17e25fc5287613693c912909cb226aa4794f26a4801c85fb49849feb476c1c5acc5ea4a69d7532d4b6b1c082effbe56f80f9fad116927564e8b029932f4e6affbe7691fc287091ca0b218f1a3f5ad66051c838bcd830060070154556e69782074696d653a2031363234383036363535c834641000000000

Reviewer: ```suggestion
INFO bridge Signed Millau Call: 0x79...00
```

> TRACE bridge Sent transaction to Millau node: 0x5e689fd157e6a30c3ed115700b7804fe7c29df4355d023996c4735374212bb010bfaacfed925b66f018ba2030d78c4f8dba2634c265a1646bb73bb51c2b4adf4

Reviewer: ```suggestion
TRACE bridge Sent transaction to Millau node: 0x5e...f4
```

Reviewer: You haven't addressed this

```

About how to interact with and test the bridge please check [send message](./docs/send-message.md) for detail explanation

> 

Reviewer: Similar to the other comment I made on this above, I would leave this at the beginning of the section so that readers have an introduction to what the section is going to cover.
